### PR TITLE
[eclipse/xtext#1812] Use centos-7 pod template & config. JDK

### PR DESCRIPTION
**Pull request by kthoms**
The centos-7 pod template provides a configuration suitable for most
builds. This makes the configuration of the custom container obsolete.

Additionally JDK versions are now selectable by parameter.

The nightly build uses 'latest' target and JDK 11.

Signed-off-by: Karsten Thoms <karsten.thoms@karakun.com>

### CODE REVIEW

**Comment by cdietrich** [comment]
do we miss a 

`wrap([$class: 'Xvnc', takeScreenshot: false, useXauthority: true]) {` ?

**Comment by kthoms** [comment]
Right, this is missing. Adding immediately.

**Comment by kthoms** [comment]
Current build failure:
```
09:07:41  Execution failed for task ':org.eclipse.xtend.core.tests:test'.
09:07:41  > Process 'Gradle Test Executor 1' finished with non-zero exit value 137
09:07:41    This problem might be caused by incorrect test process configuration.
09:07:41    Please refer to the test execution section in the User Manual at https://docs.gradle.org/5.6.3/userguide/java_testing.html#sec:test_execution
```

**Comment by kthoms** [comment]
I was able to reproduce the OOM locally.

Steps:
Execute in Xtext git root:
```
run -it -v $(pwd)/:/home/jenkins/xtext -m 4096Mi docker.io/eclipsecbi/jiro-agent-centos-7:remoting-4.2.1 bash
```

In the container execute:
```
cd xtext/xtext-xtend

export JENKINS_URL=https://ci.eclipse.org/xtext/
export GRADLE_OPTS="-Dorg.gradle.daemon=false"

./gradlew \
  clean cleanGenerateXtext build publish \
  -Dmaven.repo.local=$(pwd)/.m2/repository \
  -PcompileXtend=true \
  -PJENKINS_URL=$JENKINS_URL \
  -PignoreTestFailures=true \
  --refresh-dependencies \
  --continue
```

=> 
```
> Task :org.eclipse.xtend.core.tests:test FAILED

> Task :org.eclipse.xtend.core.tests.java8:compileTestJava
Note: /home/jenkins/xtext/xtext-xtend/org.eclipse.xtend.core.tests.java8/testdata/testdata/ClassWithVarArgs.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':org.eclipse.xtend.core.tests:test'.
> Process 'Gradle Test Executor 1' finished with non-zero exit value 137
  This problem might be caused by incorrect test process configuration.
  Please refer to the test execution section in the User Manual at https://docs.gradle.org/5.6.3/userguide/java_testing.html#sec:test_execution
```

**Comment by cdietrich** [comment]
eclipse tests seem to be killed too
https://ci.eclipse.org/xtext/job/xtext-xtend/job/kt_issue1812_parametrized_jdk/16/console

File: releng/org.eclipse.xtend.tycho.tests.parent/pom.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 	<properties>
 		<os-jvm-flags />
-		<memory-settings>-Xmx2G</memory-settings>
+		<memory-settings>-Xmx1500m</memory-settings>
```
Comment (cdietrich) [inline releng/org.eclipse.xtend.tycho.tests.parent/pom.xml:16]: i wonder why we are still oom killed.

**Review by Hejado — APPROVED** [review]

**Pull request merged**